Core/Src/main.c

```diff
@@ -124,6 +125,12 @@ int _write(int file, char* ptr, int len) {
   }
   return len;
 }
+
+/*Callback for UART*/
+void HAL_UART_TxCpltCallback(UART_HandleTypeDef *phuart) { //jake
+  HAL_UART_DMAStop(&huart3);

```

Comment:
Although this code should work, you should use *phuart (we normally just use *huart) instead of global &huart3. This change allows us to more easily change the location of this function, which I anticipate happening cuz someone will probably cause a fuss about this being in main.